src/material/button/button.md

```diff
@@ -33,10 +33,46 @@ screen-readers, which will read the text character-by-character. We leave the de
 approach this to the consuming app.
 
 ### Accessibility
-Angular Material uses native `<button>` and `<a>` elements to ensure an accessible experience by
-default. The `<button>` element should be used for any interaction that _performs an action on the
-current page_. The `<a>` element should be used for any interaction that _navigates to another
-view_.
 
+The ARIA spec gives different uses for buttons and links. In order to give the correct information to
+assistive technology, it is important to use the correct one for each situation. 
+
+ - Use the `<button>` element for any interactions that _performs an action on the
+current page_.
+ - Use the `<a>` element for links. That is iteractions that _navigates to another
+view_. This includes navigating to another page, another view in a single-page application and to elements on the same page. Include the href attribute when using the `<a>` element.
+
+#### Links That Look Like Buttons
+Generally, the `<a>` should be used for interactions that perform navigation. It ensures that assistive technology has the
+correct semantic information for the interaction. If a `<button>` must be used, then do the following.
+
+ - Give it the `role='link'` attribute.
+ - Assign an `href` attribute with the link's destination.
+
+#### Icon Buttons
 Buttons or links containing only icons (such as `mat-fab`, `mat-mini-fab`, and `mat-icon-button`)
 should be given a meaningful label via `aria-label` or `aria-labelledby`.
+
+As always, images inside buttons should contain an meaningful alt text.
+
+#### Stateful Buttons
+The mat-button supports the `disabled` attribute for indicating to the user that is cannot be interacted with. If a button needs to have binary states, such as for pressed/not pressed or on/off, consider using a Checkbox component or a Slide Toggle component. Use the Toggle Button component for interactions with more than two states.
+
+#### Binary State Buttons
+
+If a `button` must be used for an interation that has two states, then follow these guidelines.
+
+Use one of the following options to indicate the state of the button. Do not use both because it can provide confusing
+information to assistive technology.
+ - Toggle state by setting `aria-pressed="true"` and `false`
+ - Toggle state by reassigning the button label \(e.g. _on_/_off_\)
+
+Use `button` elements, not `a` elements
+
+We recommend using styles that target the aria-pressed attribute, rathen than toggling a CSS class for `pressed`/`not-pressed`.
+
+An alternative approach to make a checkbox that looks like a button using the following steps.
+
+ - Give the button a `role='checkbox'` attribute.
+ - Assign `aria-checked='true'` or `'false'` to indicate the current state.
```

Comment:
You'll also need to ensure that the psuedo-button-checkbox only activates on Enter, and not space, as native buttons do. In general, I'm a little weary about providing guidance on how to overriding elements' default roles appropriately because it's tricky to do so and also exacerbates confusing UX in applications as well as https://www.w3.org/WAI/WCAG21/Understanding/consistent-identification.html (e.g. it's better to just have checkboxes that look like checkboxes and buttons that look like buttons than to mix and match visuals and semantics).